src/Controls/src/Core/Platform/Android/TabbedPageManager.cs

```diff
@@ -633,76 +633,69 @@ protected virtual ColorStateList GetItemTextColorStates()
 
 		protected virtual ColorStateList GetItemIconTintColorState()
 		{
-			if (IsBottomTabPlacement)
-			{
-				if (_orignalTabIconColors is null)
-					_orignalTabIconColors = _bottomNavigationView.ItemIconTintList;
-			}
-			// this ensures that existing behavior doesn't change
-			else if (!IsBottomTabPlacement && BarSelectedItemColor != null && BarItemColor == null)
-				return null;
+			if (_orignalTabIconColors is null)
+				_orignalTabIconColors = IsBottomTabPlacement ? _bottomNavigationView.ItemIconTintList : _tabLayout.TabIconTint;
 
 			Color barItemColor = BarItemColor;
 			Color barSelectedItemColor = BarSelectedItemColor;
 
-			if (barItemColor == null && barSelectedItemColor == null)
+			if (barItemColor is null && barSelectedItemColor is null)
 				return _orignalTabIconColors;
 
-			if (_newTabIconColors != null)
+			if (_newTabIconColors is not null)
 				return _newTabIconColors;
 
-			int defaultColor;
+			int defaultColor = 0;
+			int checkedColor;
 
 			if (barItemColor is not null)
-			{
 				defaultColor = barItemColor.ToPlatform().ToArgb();
-			}
 			else
-			{
+				defaultColor = GetDefaultColor(defaultColor);
```

Comment:
If I am understanding the code path of the usage of the `GetDefaultColor`, if the `BarItemColor` is not set, then the value of `defaultColor` that is passed into the method will be the initial 0. In fact, since this is a local variable, it will always be 0. Is that OK?

Reply:
- I have modified the test sample to change the` icon color directly` in the button click event method. 

- I have removed the unnecessary parameter from the `GetDefaultColor` method. 

Could you please review and let us know if you have any other concerns?